src/services/ResourceBookingService.js

```diff
@@ -120,8 +121,34 @@ async function updateResourceBooking (currentUser, id, data) {
   data.updatedAt = new Date()
   data.updatedBy = await helper.getUserId(currentUser.userId)
 
-  await resourceBooking.update(data)
+  const updatedResourceBooking = await resourceBooking.update(data)
   await helper.postEvent(config.TAAS_RESOURCE_BOOKING_UPDATE_TOPIC, { id, ...data })
+  // When we are updating the status of ResourceBooking to `assigned`
+  // the corresponding JobCandidate record (with the same userId and jobId)
+  // should be updated with the status `selected`
+  if (data.status === 'assigned') {
```

Comment:
@imcaizheng This logic would work too if we are updating `resourceBooking` which is already in `assigned` status. Let's only run this logic when the status was different before, so we don't run extra updates logic for `JobCandidate`.

Or do you see any use case when it could be useful?

Reply:
@maxceem You are right. The logic should be changed.

Reply:
@maxceem updated and tested locally